src/main.js

```diff
@@ -82,69 +81,117 @@ Apify.main(async () => {
         requestQueue,
         maxRequestRetries: 3,
         handlePageTimeoutSecs: 240,
-        maxConcurrency: 20,
-        launchPuppeteerOptions: {
-            useApifyProxy: proxyConfiguration.useApifyProxy,
-            timeout: 120 * 1000,
+        maxConcurrency: 50,
+        launchContext: {
             useChrome: true,
-            stealth: true,
-            headless: true,
-            // devtools: true
+            launchOptions: {
+                headless: Apify.isAtHome(),
+                devtools: !Apify.isAtHome(),
+            }
         },
-        useSessionPool: proxyConfiguration.useApifyProxy,
-
+        browserPoolOptions: {
+            useFingerprints: true,
+        },
+        useSessionPool: true,
+        proxyConfiguration: proxyConfig,
         gotoFunction: async ({ request, page }) => {
             await Apify.utils.puppeteer.blockRequests(page, {
-                extraUrlPatterns: [ 'brain.foresee.com*', 'c.amazon-adsystem.com*' ]
+                extraUrlPatterns: ['brain.foresee.com*', 'c.amazon-adsystem.com*'],
             });
 
             return page.goto(request.url, {
                 timeout: 180 * 1000,
-                waitUntil: 'load' // networkidle0
+                waitUntil: 'load', // networkidle0
             });
         },
 
         handlePageFunction: async ({ page, request, session }) => {
             // if exists, check items limit. If limit is reached crawler will exit.
             if (maxItems) maxItemsCheck(maxItems, itemCount);
 
-            log.info('Processing:', request.url);
-            const { label } = request.userData;
+            log.info('Processing:', { url: request.url });
+            let { label } = request.userData;
+
+            if (label === 'FILTER') {
+                const searchUrl = request.url;
+                const splitUrl = searchUrl.split('?')[0];
+                const qs = querystring.parse(searchUrl.split('?')[1]);
+                const minPrice = Number(qs.minPrice) ? Number(qs.minPrice) : 0;
+                const maxPrice = Number(qs.maxPrice) ? Number(qs.maxPrice) : 10000000;
+                await puppeteer.injectJQuery(page);
+
+                const totalResults = await page.evaluate(() => Number(
+                    $('span:contains(Results)')
+                        .text()
+                        .replace(/\D/g, ''),
+                ),
+                );
+
+                // split urls by price filter
+                if (totalResults > 1000) {
+                    log.info(
+                        'Too many results for one scrape, splitting the search into two.',
```

Comment:
Logs should always include info about the current URL and you could even add how you are gonna split it.